Figure out "dtslint" problem

Seems it tries to test TypeScript types against "next" versions of TypeScript and keeps breaking

Getting error
```
Error: At /Users/gleb/git/cypress/cli/node_modules/@types/sinon/index.d.ts:{"line":13,"character":3}: 'tslint:disable' is forbidden. ('tslint:disable:rulename', tslint:disable-line' and 'tslint:disable-next-line' are allowed.)
```

ok, we are using `@types/sinon: 4.0.0` types for sinon.js but it does not support TypeScript v3, looking at NPM labels

```
$ vers @types/sinon
4.3.3          7 months   ts2.3, ts2.4, ts2.5, ts2.6, ts2.7
7.0.0          6 days     latest, ts2.8, ts2.9, ts3.0, ts3.1, ts3.2, ts3.3
```

Ok, so our version of Sinon.js types is old, and we need to update because TypeScript 3.x is good and we should not fall behind.

Let's see - in `cli` package we have prod `@types/sinon` and dev `sinon`. Trying to upgrade them to latest. Getting now TS errors for TypeScript 2.5

```
$ npm run dtslint

> cypress@0.0.0 dtslint /Users/gleb/git/cypress/cli
> dtslint types

Error: Errors in typescript@2.5 for external dependencies:
../node_modules/@types/sinon/index.d.ts(20,19): error TS2300: Duplicate identifier 'Sinon'.
../node_modules/@types/sinon/index.d.ts(1448,36): error TS1005: ';' expected.
```

Upgraded min version in our types to match sinon JS min version

--- a/cli/package.json
+++ b/cli/package.json
@@ -23,7 +23,7 @@
     "test-cov": "nyc npm run unit",
     "unit": "BLUEBIRD_DEBUG=1 NODE_ENV=test bin-up mocha --reporter mocha-multi-reporters --reporter-options configFile=../mocha-reporter-config.json",
     "lint": "bin-up eslint --fix *.js bin/* lib/*.js lib/**/*.js test/*.js test/**/*.js",
-    "dtslint": "echo 'disabling dtslint for now'",
+    "dtslint": "dtslint types",
     "prebuild": "npm run test-dependencies && node ./scripts/start-build.js",
     "build": "node ./scripts/build.js",
     "prerelease": "npm run build",
@@ -48,7 +48,7 @@
     "@types/lodash": "4.14.87",
     "@types/minimatch": "3.0.3",
     "@types/mocha": "2.2.44",
-    "@types/sinon": "4.0.0",
+    "@types/sinon": "7.0.0",
     "@types/sinon-chai": "2.7.29",
     "bluebird": "3.5.0",
     "cachedir": "1.3.0",
@@ -96,7 +96,7 @@
     "nyc": "13.0.0",
     "proxyquire": "2.0.1",
     "shelljs": "0.7.8",
-    "sinon": "5.0.7",
+    "sinon": "7.2.2",
     "snap-shot-it": "^5.0.0"
   },
   "files": [

--- a/cli/types/index.d.ts
+++ b/cli/types/index.d.ts
@@ -4,7 +4,7 @@
 //                 Mike Woudenberg <https://github.com/mikewoudenberg>
 //                 Robbert van Markus <https://github.com/rvanmarkus>
 //                 Nicholas Boll <https://github.com/nicholasboll>
-// TypeScript Version: 2.5
+// TypeScript Version: 2.8
 // Updated by the Cypress team: https://www.cypress.io/about/
 
 /// <reference path="./blob-util.d.ts" />
@@ -3926,8 +3926,8 @@ declare namespace Cypress {
 
   // Diff / Omit taken from https://github.com/Microsoft/TypeScript/issues/12215#issuecomment-311923766
   type Diff<T extends string, U extends string> = ({[P in T]: P } & {[P in U]: never } & { [x: string]: never })[T]
-  // @ts-ignore TODO - remove this if possible. Seems a recent change to TypeScript broke this. Possibly https://github.com/Microsoft/TypeScript/pull/17912
-  type Omit<T, K extends keyof T> = Pick<T, Diff<keyof T, K>>
+  // TODO - remove this if possible. Seems a recent change to TypeScript broke this. Possibly https://github.com/Microsoft/TypeScript/pull/17912
+  type Omit<T, K extends keyof T> = Pick<T, Diff<keyof T, K>> // tslint:disable-line
 }
 
 /**

